corehq/apps/domain/extension_points.py

```diff
@@ -27,6 +27,8 @@ def validate_password_rules(password) -> Optional[str]:
     Returns:
         An error message to show
     """
+    if len(password) < settings.MINIMUM_PASSWORD_LENGTH:
+        return _(f"Password should have at least {settings.MINIMUM_PASSWORD_LENGTH} characters.")
```

Comment:
This won't end up getting translated, because `_` can't handle dynamic strings. The way to do this is to translate, then substitute, using `format`: `_("Password should have at least {} characters.").format(settings.MINIMUM_PASSWORD_LENGTH)`

Reply:
Nit: it would be nice if the error message text was consistent everywhere (this might also simplify the translation files, although I'm not sure about this). In some of the other messages you say _"Password **must** have ..."_, which I think is clearer to the end user than saying "should".

(Also, thanks @orangejenny for clarifying that f-strings don't work here, I wasn't sure about this myself).

Reply:
It will simplify the translation files - identical strings need to only be translated once by the translators.  The `.po` files will just reference each occurrence above the string.

Regarding the f-strings, I just wrote up a little explainer of how translations work that might help clarify this type of questions.  Would love some feedback: https://github.com/dimagi/commcare-hq/pull/31853 (readable version here: [How translations work](https://github.com/dimagi/commcare-hq/blob/602dc85e8e4ccde35b4f16ef42e44621946c09bd/docs/translations.rst#how-translations-work))

Reply:
@orangejenny  Hi Jenny, thanks for pointing this out!!
@esoergel Hi Ethan, thanks for the doc!! The example list is excellent!

Reply:
@joxl You're right. It will simplify the translation files because they'll all share the same `msgid`.